Source/Core/CoplanarPolygonGeometryLibrary.js

```diff
@@ -0,0 +1,131 @@
+define([
+        './defined',
+        './Cartesian2',
+        './Cartesian3',
+        './Check',
+        './IntersectionTests',
+        './Math',
+        './Matrix3',
+        './OrientedBoundingBox',
+        './Plane',
+        './Ray'
+    ], function(
+        defined,
+        Cartesian2,
+        Cartesian3,
+        Check,
+        IntersectionTests,
+        CesiumMath,
+        Matrix3,
+        OrientedBoundingBox,
+        Plane,
+        Ray
+    ) {
+    'use strict';
+
+    /**
+     * @private
+     */
+    var PolygonGeometryLibrary = {};
+
+    var scratchIntersectionPoint = new Cartesian3();
+    var scratchXAxis = new Cartesian3();
+    var scratchYAxis = new Cartesian3();
+    var scratchZAxis = new Cartesian3();
+    var scratchOrigin = new Cartesian3();
+    var scratchNormal = new Cartesian3();
+    var scratchRay = new Ray();
+    var scratchPlane = new Plane(Cartesian3.UNIT_X, 0);
+    var obbScratch = new OrientedBoundingBox();
+
+    // call after removeDuplicates
+    PolygonGeometryLibrary.projectTo2D = function(positions, positionsResult, normalResult, tangentResult, bitangentResult) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.defined('positions', positions);
+        Check.defined('positionsResult', positionsResult);
+        //>>includeEnd('debug');
+
+        var obb = OrientedBoundingBox.fromPoints(positions, obbScratch);
+        var halfAxes = obb.halfAxes;
+        var xAxis = Matrix3.getColumn(halfAxes, 0, scratchXAxis);
+        var yAxis = Matrix3.getColumn(halfAxes, 1, scratchYAxis);
+        var zAxis = Matrix3.getColumn(halfAxes, 2, scratchZAxis);
+
+        var xMag = Cartesian3.magnitude(xAxis);
+        var yMag = Cartesian3.magnitude(yAxis);
+        var zMag = Cartesian3.magnitude(zAxis);
+        var min = Math.min(xMag, yMag, zMag);
+
+        var i;
+        // If all the points are on a line, just remove one of the zero dimensions
+        if ((xMag === 0 && (yMag === 0 || zMag === 0)) || (yMag === 0 && zMag === 0)) {
```

Comment:
Yep, we decided a while ago to just return undefined and not render invalid geometry instead of throwing an error.  This is mostly because there's a lot of garbage data in KML and GeoJSON files that are automatically generated and we don't want to crash the whole file because we don't know how to render one of the polygons